lib/mongoid/criteria/queryable/extensions/date.rb

```diff
@@ -25,7 +25,7 @@ def __evolve_date__
           #
           # @return [ Time ] The date as a local time.
           def __evolve_time__
-            ::Time.local(year, month, day)
+            ::Time.configured.local(year, month, day).to_time
```

Comment:
I've added .to_time here in order to preserve the return time. However, it's not really necessary; AS::TimeWithZone works equivalently. `__mongoize_time__` will return either Time or AS::TimeWithZone based on Time.configured. Please let me know if you'd like me to remove this `to_time` to make it consistent with `__mongoize_time__`.